app/ui/src/network.ts

```diff
@@ -216,8 +216,8 @@ class Network {
                 }
             }
 
-            xhr.onerror = () => {
-                reject('Network Failure')
+            xhr.onerror = (event) => {
+                reject(new Error(`XHR request failed: ${event}\n Type: ${event.type}: ${event.loaded} bytes transferred`))
```

Comment:
Shouldn't this be NetworkError?

Reply:
58bf3c26f5e6a0c12f4d65d7de40629a3a5a40aa